src/EnergyPlus/WaterCoils.cc

```diff
@@ -1763,6 +1763,8 @@ void InitWaterCoil(EnergyPlusData &state, int const CoilNum, bool const FirstHVA
                 CalcSimpleHeatingCoil(state, CoilNum, HVAC::FanOp::Continuous, 1.0, state.dataWaterCoils->SimCalc);
                 coilTypeName = "Coil:Heating:Water";
             }
+            state.dataRptCoilSelection->coilSelectionReportObj->setCoilEqNum(
+                state, waterCoil.Name, coilTypeName, state.dataSize->CurSysNum, state.dataSize->CurOASysNum, state.dataSize->CurZoneEqNum);
```

Comment:
Interesting find on water coils. The sizer functions are not called for a hard-sized coils and therefore the index to the zone, system, or OA system is never initialized. So when the coil reports loops through all coils and looks for `c->zoneEqNum > 0` it skips hard-sized water coils.